dashboard/app/models/script_level.rb

```diff
@@ -542,4 +546,39 @@ def hidden_for_section?(section_id)
   def should_hide_survey(user, viewed_user)
     anonymous? && user.try(:teacher?) && !viewed_user.nil? && user != viewed_user
   end
+
+  def seeding_key(seed_context, use_existing_level_keys = true)
```

Comment:
previous discussion suggested we might just use a GUID for this. Was there an explicit decision to stick with this implementation which piggy-backs on existing fields? I can see arguments for both sides.

Reply:
I think we definitely want to use GUIDs for any new models we add. They should still implement the `seeding_key(seed_context)` interface, but they can simply return their GUID.

I'm undecided on whether we want to backfill GUIDs for all existing models. I was partway through this implementation when we started discussing it, so I wanted to finish it out first to have it as an option. I think we can consider backfilling GUIDS for existing models to be optional followup work - it doesn't need to happen for this PR to be merged.

Thoughts?

Reply:
The trade-offs I'm aware of with moving to GUIDs for any particular model are:

* PRO: eliminate complexity of seeding_key methods which factor in more than ~2 inputs
* CON: cost of back-filling existing models

are there any others you're aware of that I'm missing?

Reply:
I'm also thinking that there'll be a cost in implementing logic to assign a GUID to any new object that's created. But maybe that's actually easy - with a callback on create of some kind?

Other than that, not aware of anything beyond what you listed.

Reply:
> I'm also thinking that there'll be a cost in implementing logic to assign a GUID to any new object that's created. But maybe that's actually easy - with a callback on create of some kind?

yes, a block passed to `create` should do it nicely. If used with `find_or_create_by`, the block only gets called on create. you can even use the block to supply parameters that are required fields in the database: 
https://github.com/code-dot-org/code-dot-org/blob/bab074cc8c2e10ab5736711c56bd599c8f15379e/dashboard/app/models/unit_group.rb#L173-L175
https://github.com/code-dot-org/code-dot-org/blob/1f773db26d9279a517dbb3251d53ba2d8f1c3704/dashboard/app/models/unit_group_unit.rb#L8

Reply:
I just realized this exact option may not work with activerecord import. so, maybe you were on the right track with callbacks.

Reply:
^ That might still be fine since anything we import should already have a GUID. We just need to make sure that anywhere else in the code that creates new objects creates the GUID.

If we had to supply a block each time though I'd be concerned that we'd miss some cases.

Reply:
if the DB column is not nullable and has a unique index, or if a validation checks for its presence + uniqueness, then I'd feel comfortable enough that future developers would not forget to specify it.